### PR TITLE
elmerfem: init at release-8.4

### PR DESCRIPTION
**Pull request by Wulfsta**
###### Motivation for this change

Add Elmer FEM to Science/Physics.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Wulfsta** [comment]
Cleaned up commits to align with contributing guidelines.

**Comment by Wulfsta** [comment]
@GrahamcOfBorg build elmerfem

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8247,6 +8247,12 @@
     githubId = 483465;
     name = "Mateusz Wykurz";
   };
+  wulfsta = {
+    email = "wulfstawulfsta@gmail.com";
+    github = "Wulfsta";
+    githubId = 13378502;
+    name = "Wulfsta";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:8255]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/applications/science/physics/elmerfem/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  nativeBuildInputs = [ cmake pkgconfig git ];
+  #nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ gfortran openmpi blas liblapack qt4 qwt6_qt4 ];
+
+  preConfigure = ''
+    patchShebangs ./
+  '';
+
+  storepath = (placeholder "out");
```
Comment (jonringer) [inline pkgs/applications/science/physics/elmerfem/default.nix:27]:
```suggestion
  storepath = placeholder "out";
```

File: pkgs/applications/science/physics/elmerfem/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    owner = "elmercsc";
+    repo = "elmerfem";
+
+    rev = "release-${version}";
+    sha256 = "0vk31lplxlng173q8jjcpbyj1gaf98jvkqjvi9077d1nslya7vpm";
+    #rev = "b2a9134a9d3d883f733f5053e5db317b24211db2";
+    #sha256 = "0xgrpa1016rdq8qch8rhs91k9zrk3dh86rg14cz505x4ni5g4488";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig git ];
```
Comment (jonringer) [inline pkgs/applications/science/physics/elmerfem/default.nix:19]: pkg-config is canonical name, pkgconfig is an alias
```suggestion
  nativeBuildInputs = [ cmake pkg-config git ];
```

File: pkgs/applications/science/physics/elmerfem/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+stdenv.mkDerivation rec {
+  pname = "elmerfem";
+  version = "8.4";
+
+  hardeningDisable = [ "format" ];
+
+  src = fetchFromGitHub {
+    owner = "elmercsc";
+    repo = "elmerfem";
+
```
Comment (jonringer) [inline pkgs/applications/science/physics/elmerfem/default.nix:12]:
```suggestion
```

File: pkgs/applications/science/physics/elmerfem/default.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    rev = "release-${version}";
+    sha256 = "0vk31lplxlng173q8jjcpbyj1gaf98jvkqjvi9077d1nslya7vpm";
+    #rev = "b2a9134a9d3d883f733f5053e5db317b24211db2";
+    #sha256 = "0xgrpa1016rdq8qch8rhs91k9zrk3dh86rg14cz505x4ni5g4488";
```
Comment (jonringer) [inline pkgs/applications/science/physics/elmerfem/default.nix:16]:
```suggestion
```

File: pkgs/applications/science/physics/elmerfem/default.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
```
Comment (jonringer) [inline pkgs/applications/science/physics/elmerfem/default.nix:18]: src usually comes after version, hardening is usually after src
```suggestion
  src = fetchFromGitHub {
    owner = "elmercsc";
    repo = "elmerfem";

    rev = "release-${version}";
    sha256 = "0vk31lplxlng173q8jjcpbyj1gaf98jvkqjvi9077d1nslya7vpm";
    #rev = "b2a9134a9d3d883f733f5053e5db317b24211db2";
    #sha256 = "0xgrpa1016rdq8qch8rhs91k9zrk3dh86rg14cz505x4ni5g4488";
  };
  
  hardeningDisable = [ "format" ];
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Wulfsta) [reply maintainers/maintainer-list.nix:8255]: How do I go about doing this?

**Comment by Wulfsta** [comment]
Split commit and added changes, probably not in the ideal way.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (jonringer) [reply maintainers/maintainer-list.nix:8255]: looks like you figured this out, but the easiest way I can think of:

```
git reset HEAD^ -- maintainers/
git add maintainers
git commit -m "maintainers: add <name>"
git rebase -i HEAD^^ # switch order, save
```

File: pkgs/applications/science/physics/elmerfem/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    owner = "elmercsc";
+    repo = "elmerfem";
+    rev = "release-${version}";
+    sha256 = "0vk31lplxlng173q8jjcpbyj1gaf98jvkqjvi9077d1nslya7vpm";
+  };
+
+  hardeningDisable = [ "format" ];
+
+  nativeBuildInputs = [ cmake pkg-config git ];
+  #nativeBuildInputs = [ autoreconfHook ];
```
Comment (jonringer) [inline pkgs/applications/science/physics/elmerfem/default.nix:17]:
```suggestion
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (Wulfsta) [reply maintainers/maintainer-list.nix:8255]: Yeah I just checked out a new branch following upstream, made the changes there, then reverted the master branch and merged the new one.

File: pkgs/applications/science/physics/elmerfem/default.nix (hunk 7 above)
Comment (Wulfsta) [reply pkgs/applications/science/physics/elmerfem/default.nix]: Huh, I though I had fixed this. Must have been working in the wrong branch.

**Comment by jonringer** [comment]
please squash the last two commits:

```
git reset HEAD^
git add pkgs/
git commit --amend --no-edit
git push .. .. --force
```

**Comment by Wulfsta** [comment]
@jonringer Squashed, could you take a look at this when you have a chance? Thanks!

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

```
https://github.com/NixOS/nixpkgs/pull/85962
1 package built:
elmerfem
```

**Comment by jonringer** [comment]
Congrats on your first PR @Wulfsta 

Sorry, i was busy :)

**Comment by Wulfsta** [comment]
@jonringer Thanks, there will be another shortly! And don't apologize, there was no rush!
